mdformat/renderer/__init__.py

```diff
@@ -142,5 +89,125 @@ def _write_references(env: dict) -> str:
             if title:
                 title = title.replace('"', '\\"')
                 item += f' "{title}"'
-            text += item + "\n"
-        return text.rstrip()
+            ref_list.append(item)
+        return "\n".join(ref_list)
+
+
+class SyntaxTreeNode:
```

Comment:
so this is just a reworking of `NestTokens` and `nest_tokens` (see https://github.com/executablebooks/markdown-it-py/blob/2afb7cc5a824cf8340e405704cc996208ae33d0e/markdown_it/token.py#L114). I think it should just go into markdown-it-py and deprecate those

Reply:
Yeah precisely so. Why I didn't use `NestedTokens` is explained in this closed draft PR https://github.com/executablebooks/markdown-it-py/pull/117

> I think it should just go into markdown-it-py and deprecate those

That would be great, but it will mean some more work for you :smile: I understood `NestedTokens` is a central piece in MyST parser. But I'm very much willing to move this class to `markdown_it` if you think its a good idea. I think the `.render` method should probably be removed then, and I'll subclass and add it back here in `mdformat`. That method is quite `mdformat` and rendering specific...

Reply:
> I understood `NestedTokens` is a central piece in MyST parser

yes but we can just deprecate it, leaving it in the code base for now, and I can migrate in due course